reference_data/management/tests/update_dbnsfp_gene_tests.py

```diff
@@ -0,0 +1,45 @@
+
+dbNSFP_GENE_DATA = [
+    "OR4F5	ENSG00000186092	1	.	.	Q8NH21	OR4F5_HUMAN	79501	CCDS30547	NM_001005484	uc001aal.1	.	618355	olfactory receptor family 4 subfamily F member 5	.	.	.	Olfactory transduction - Homo sapiens (human);Olfactory receptor activity;Signaling by GPCR;Signal Transduction;Olfactory Signaling Pathway;G alpha (s) signalling events;GPCR downstream signalling	hsa04740	Olfactory transduction	FUNCTION: Odorant receptor. {ECO:0000305}.; 	.	.	.	.	.	.	.	G protein-coupled receptor signaling pathway;detection of chemical stimulus involved in sensory perception of smell	plasma membrane;integral component of membrane	G protein-coupled receptor activity;olfactory receptor activity	.	.	.	1	1	1	0.06092	0.186891868710518	N	.	0.07159	.	.	.	0.9905209	.	.	0.176329298172162	0.644086264144236	0.179584437683602	0.550302420215064	0.39663970580189	0.0530578739830465	0.179613130021276	0.623408532464667	0.196978337514057	3.0650e-02	6.1116e-01	3.5819e-01	.	.	.	.	113.16669	2.31572	Medium	Medium	Medium	Medium	Medium	Medium	Medium	Medium	Medium	Medium	.	3.9936102236421724E-4	0.0	3.9936102236421724E-4	0.0	0.022763578274760384	0.00439297124600639	0.008386581469648562	0.001996805111821086	.	.	.	N	0.114371359811310	N	.	.	.	.	.	.\n",
+    "MIR1302-2HG	.	.	.	.	.	107985730	.	XR_001737835	.	.	618355	MIR1302-2 host gene	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	1	1	1	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	0.0	0.0	0.0	0.0	0.0	0.0	0.0	0.0	.	.	.	.	.	.	.	.	.	.	.	.\n",
+    "A1BG-AS1	ENSG00000268895	19	NCRNA00181;A1BGAS;A1BG-AS	FLJ23569	.	.	503538	.	NR_015380	uc002qse.3	.	.	A1BG antisense RNA 1	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	1	1	1	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	7.987220447284345E-4	0.0	7.987220447284345E-4	0.0	0.02875399361022364	0.0	0.017172523961661343	0.0	.	.	.	.	.	.	.	.	.	.	.	.\n"
+]
+
+
+class UpdateDbNsfpGeneTest(TestCase):
+    fixtures = ['users', 'reference_data']
+    multi_db = True
+
+    @mock.patch('reference_data.management.commands.utils.update_utils.logger')
+    @mock.patch('reference_data.management.commands.utils.update_utils.tqdm')
+    @mock.patch('os.path.isfile')
+    @mock.patch.object(__builtin__, 'open')
+    @mock.patch('reference_data.management.commands.utils.download_utils.download_file')
+    def test_update_dbnsfp_gene_command(self, mock_download, mock_open, mock_isfile, mock_tqdm, mock_logger):
+        mock_isfile.return_value = False
+        mock_download.return_value = 'dbNSFP_gene'
+        f = mock_open.return_value.__enter__.return_value
+        f.readline.return_value = dbNSFP_HEADER
+        mock_tqdm.return_value = dbNSFP_GENE_DATA
```

Comment:
tqdm is a wrapper library that return the given iterable and prints some stuff about it. Don't mock it, as was discussed in previous PRs. If the print output is really messing with logging tests, then use a side_effect so it just returns whatever iterable it is given. But having a printing utility totally change the given value is really unintuitive and dpes not do a good job testing the given files

Reply:
Okay